_episodes/24.md

```diff
@@ -0,0 +1,412 @@
+   system. The isolation restricts any interference from other processes. 
+   Containers achieve the goals of virtualization without sacrificing much 
+   performance or efficiency. 
+   
+
+   <p align="center">
+    <img align="center" width="75%" height="100%" src="/assets/episode/24/container-evolution.svg"/><br/>
+    Source: https://kubernetes.io/docs/concepts/overview/what-is-kubernetes/
+   </p>
+   
+   - Containerization simplified a trend in service architecture called 
+   microservices. Microservices deploy loosely coupled and modular applications
+   rather than all-encompassing monolithic applications. With containers, these
+   applications can be deployed and scaled up quickly across various virtual and
+   physical machines without affecting applications on the same machine. This
+   is great, but results in new complexities. Some examples are the need for 
+   new approaches to monitoring the health of applications, scaling the 
+   applications as requests grow and diminish, redeploying crashed application, 
+   and networking the applications together. In summary, all of these activities
+   can be considered container orchestration and this is exactly what Kubernetes
+   solves!
+   
+   <p align="center">
+    <img align="center" width="75%" height="100%" src="/assets/episode/24/load-balancer.jpeg"/><br/>
+    Source: https://www.slideshare.net/devopsdaysaustin/continuously-delivering-microservices-in-kubernetes-using-jenkins<br/>
+    Here we hae two services that each sit behind a load balancer provided and mapped by the Kuberenets cluster.
+   </p>
+
+1. **Kubernetes components and architecture**:
+
+   - Node - The physical machine or VM running a kubelet and container runtime.
+   - Control Plane - The container orchestration layer that exposes the API and 
+   interfaces to define, deploy, and manage the lifecycle of containers.
+   - Cluster - a set of nodes connected to the same control plane.
+   - Pod - single instance of an application, the smallest object in kubernetes.
+
+   <p align="center">
+    <img align="center" width="75%" height="100%" src="/assets/episode/24/components-of-kubernetes.svg"/><br/>
+    Source: https://kubernetes.io/docs/concepts/overview/components/
+   </p>
+
+Kubernetes control plane components:
+
+ - API server that nodes connect with and is the front end for users and 
+ administrators of the cluster.
+ - etcd keystore is a distributed store containing all data used to manage 
+ the cluster
+ - Scheduler that distributes work across nodes and assigns newly created 
+ containers to nodes
+ - Controllers that are the brain behind orchestration and monitors for 
+ nodes going down etc...
+     
+Kubernetes worker nodes components:
+ - container runtime - is the underlying runtime used to manage containers 
+ - kubelet - agent that checks the health and manages the pods running on the node based on the desired state provided in the PodSpec
```

Comment:
I don't think PodSpec has been introduced before this point? No idea if it matters or not.

Reply:
If anything hasn't been covered we point to the kubernetes documentation